libs/category/src/filters/type/equal/behaviors/apply/apply.ts

```diff
@@ -0,0 +1,27 @@
+import { DaffCategoryEqualFilter } from '../../../../../models/public_api';
+import {
+  DaffCategoryFilterEqualRequest,
+  DaffToggleCategoryFilterEqualRequest,
+} from '../../../../../models/public_api';
+
+/**
+ * Applies a filter request to a group of equal type filter options .
+ */
+export const daffApplyFilterEqual = (
+  request: DaffCategoryFilterEqualRequest,
+  filter: DaffCategoryEqualFilter,
+): DaffCategoryEqualFilter => ({
+  ...filter,
+  options: request.value.reduce((acc, value) => {
+    const option = filter.options[value];
+
+    if (option) {
+      acc[value] = {
```

Comment:
This doesn't really make sense to me. The key for the dictionary entry of `options` is the `value` field of the entry itself? Adding docs to the `DaffCategoryFilterEqualOption` model and the fields inside it would help clarify this.

Reply:
The key must be the value to ensure that it is unique

Reply:
Oh, I missed the part where `DaffCategoryEqualFilter extends DaffCategoryFilterBase`. I thought you were losing the filter name information, but its just on the `DaffCategoryFilterBase`

Reply:
The models are slightly convoluted because we had to use base types and union types to workaround TS's inability to infer some types based on the usage